packages/common/infra/src/blocksuite/initialization/middleware.ts

```diff
@@ -0,0 +1,103 @@
+// TODO: remove this file after blocksuite exposed it
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import { assertExists } from '@blocksuite/global/utils';
+import type { JobMiddleware } from '@blocksuite/store';
+
+export const replaceIdMiddleware: JobMiddleware = ({ slots, workspace }) => {
```

Comment:
if we do not change the id, what problem will happen?

Reply:
For now in idb all page docs will be stored in the same table with the key = pageid. This could be a disaster if we do not replace the ids.

Reply:
ok

Reply:
Thanks! This has been fixed. Also submitted an upstream PR:
https://github.com/toeverything/blocksuite/pull/5883